### PR TITLE
Add rb_thread_call* and rb_protect functions

### PR DESCRIPTION
**Pull request by dovahcrow**

### CODE REVIEW

**Comment by d-unsed** [comment]
Hi @doomsplayer!

Thank you for your work, it will be a very useful feature!

File: src/types.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
-use libc::{c_void, intptr_t, uintptr_t};
+use libc::{intptr_t, uintptr_t};
 
-pub use libc::{c_char, c_int, c_long};
+pub use libc::{c_void, c_char, c_int, c_long};
```
Comment (d-unsed) [inline src/types.rs]: Put `c_void` to the end of the list (preserve alphabetical order).

File: src/vm.rs [hunk 2]
```diff
@@ -1,7 +1,17 @@
-use types::{c_char, Value};
+use types::{c_char, Value, CallbackPtr, c_void, c_int};
```
Comment (d-unsed) [inline src/vm.rs]: Same here, arrange the variable in alphabetical order.

**Comment by d-unsed** [comment]
Also squash commits and remove `feat(vm):` prefix from commit message.

Please change these minor things and let's merge it :)

**Comment by dovahcrow** [comment]
hi @d-unseductable, IMO these are different functions and I don't think commits should be squashed.

**Comment by d-unsed** [comment]
Ah yes, @doomsplayer that makes sense. In that case I prefer to have separate PRs. It helps to easily make changes/additions to the code related to each of the commits and keep git history clean and simple :)

Anyways do you want to split the commits back or let's merge?

**Comment by steveklabnik** [comment]
:+1: for merging either way

**Comment by dovahcrow** [comment]
@steveklabnik @d-unseductable split done.

**Comment by d-unsed** [comment]
Thank you! I will release new version to crates.io soon

**Comment by d-unsed** [comment]
@doomsplayer, published! Please update to `0.2.7`

**Comment by dovahcrow** [comment]
Thanks

**Pull request merged**